src/flb_pack.c

```diff
@@ -923,13 +924,23 @@ flb_sds_t flb_pack_msgpack_to_json_format(const char *data, uint64_t bytes,
             case FLB_PACK_JSON_DATE_JAVA_SQL_TIMESTAMP:
             /* Format the time, use microsecond precision not nanoseconds */
                 gmtime_r(&tms.tm.tv_sec, &tm);
-                s = strftime(time_formatted, sizeof(time_formatted) - 1,
+                s = strftime(time_formatted, sizeof(time_formatted),
                              FLB_PACK_JSON_DATE_JAVA_SQL_TIMESTAMP_FMT, &tm);
 
+                max_len = sizeof(time_formatted) - s;
```

Comment:
I would prefer it if instead of setting `max_len` here you set `max_len` to `sizeof(time_formatted)` before line 927, used the variable as the argument in the `strftime` call and only subtract `s` from it in line 930.